aws-api/src/test/java/com/amplifyframework/api/aws/AWSApiPluginTest.java

```diff
@@ -145,7 +143,7 @@ public void graphQlQueryRendersValidResponse() throws ApiException {
 
         GraphQLResponse<Iterable<BlogOwner>> actualResponse =
             Await.<GraphQLResponse<Iterable<BlogOwner>>, ApiException>result(((onResult, onError) ->
-                plugin.query(BlogOwner.class, onResult, onError)
+                plugin.query(ModelQuery.list(BlogOwner.class), onResult, onError)
```

Comment:
`ModelQuery` is a good name, based on what the class does.

However, when a customer uses it, they might prefer to write like:

```
api.query(Where.modelClass(BlogOwner.class), ..., ...)
```

 or something like what we did for the `QueryOptions`.

Maybe just,

```
api.query(Models.ofType(BlogOwner.class)
```

That way we at least have a separate `Models.java` to put factory methods into, that know about `<T extends Model>` (I forget where it landed, but `Where` _doesn't_ know about `<T extends Model>`, right?

Reply:
I think there's a difference here between that solution in DataStore and this one in API. Here we're not trying to hide the fact that it's GraphQL behind the scenes. Actually `Amplify.API.mutate()` and `Amplify.API.query()` makes that very explicit.

Therefore, the factory methods here resemble the GraphQL statements:

`ModelQuery.get(Todo.class)` is `query getTodo()`
`ModelMutation.create(todo)` is `mutation createTodo({})`
etc

With static imports the syntax becomes even closer:

```java
Amplify.API.query(get(Todo.class)); // query getTodo()
Amplify.API.mutate(update(todo)); // mutation updateTodo()
```

The PR description in the iOS repo also gives an idea of the experience there and how this Android builders were design to follow that same path: https://github.com/aws-amplify/amplify-ios/pull/470